keanu-project/src/main/java/io/improbable/keanu/tensor/TensorShape.java

```diff
@@ -182,5 +182,24 @@ public static boolean isScalar(int[] shape) {
         return newShape;
     }
 
+    /**
+     * Writes a lower rank shape over a higher rank shape, starting from the right.
+     *
+     * e.g: high rank shape = [1, 2, 2, 1]
+     *      low rank shape = [1, 4]
+     *
+     * Result after copy = [1, 2, 1, 4]
+     *
+     * @param higherRankShape source shape that will get written over
+     * @param lowerRankShape  shape to write
+     * @return the high rank  shape with the lower rank shape inserted on top of it
+     */
+    public static int[] copyLowRankOverHighRank(int[] higherRankShape, int[] lowerRankShape) {
```

Comment:
This only works for lowerRank to the tail end of the higher rank shape?

Reply:
Yep lower rank to the tail end of higher rank